File: programers_Lv1/숫자 문자열과 영단어.js
// https://school.programmers.co.kr/learn/courses/30/lessons/81301
function solution(s) {
  var answer = s;

  const words = [
    "zero",
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight",
    "nine",
  ];

  words.forEach((v, i) => {
    answer = answer.replaceAll(v, i); // replace 사용하면 5, 7, 8 통과 안됨
  });
  return ~~answer;
}

console.log(solution("one4seveneight"));

File: programers_Lv1/숫자 문자열과 영단어.py
# https://school.programmers.co.kr/learn/courses/30/lessons/81301?language=python3
def solution(s):
    answer = s
    words = [
        "zero",
        "one",
        "two",
        "three",
        "four",
        "five",
        "six",
        "seven",
        "eight",
        "nine",
    ]

    for i, v in enumerate(words):
        if v in s:
            answer = answer.replace(v, str(i))
    return int(answer)


print(solution("one4seveneight"))  # 1478
